### PR TITLE
perf(ivy): avoid generating extra parameters for host property bindings

### PR DESCRIPTION
**Pull request by crisbeto**
Currently we reuse the same instruction both for regular property bindings and property bindings on the `host`. The only difference between the two is that when it's on the host we shouldn't support inputs. We have an optional parameter called `nativeOnly` which is used to differentiate the two, however since `nativeOnly` is preceded by another optional parameter (`sanitizer`), we have to generate two extra parameters for each host property bindings every time (e.g. `property('someProp', 'someValue', null, true)`).

These changes add a new instruction called `hostProperty` which avoids the need for the two parameters by removing `nativeOnly` which is always set and it allows us to omit `sanitizer` when it isn't being used.

These changes also remove the `nativeOnly` parameter from the `updateSyntheticHostBinding` instruction, because it's only generated for host elements which means that we can assume that its value will always be `true`.

### CODE REVIEW

File: packages/compiler/src/render3/view/compiler.ts [hunk 1]
```diff
@@ -644,7 +644,7 @@ function createHostBindingsFunction(
   const attributeBindings: o.Expression[][] = [];
   const syntheticHostBindings: o.Expression[][] = [];
 
-  (bindings || []).forEach((binding: ParsedProperty) => {
+  bindings && bindings.forEach((binding: ParsedProperty) => {
```
Comment (pkozlowski-opensource) [inline packages/compiler/src/render3/view/compiler.ts]: I think that at this point would be to more readable to have the explicit `if` around `bindings`...

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM!

File: packages/core/src/render3/instructions/host_property.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * @codeGenApi
+ */
+export function ɵɵhostProperty<T>(
+    propName: string, value: T, sanitizer?: SanitizerFn | null): TsickleIssue1009 {
+  const index = getSelectedIndex();
+  ngDevMode && assertNotEqual(index, -1, 'selected index cannot be -1');
+  const lView = getLView();
+  const bindReconciledValue = bind(lView, value);
+  if (bindReconciledValue !== NO_CHANGE) {
+    elementPropertyInternal(index, propName, bindReconciledValue, sanitizer, true);
```
Comment (AndrewKushnir) [inline packages/core/src/render3/instructions/host_property.ts]: I'm wondering if make sense to extract a part of the logic (guarded with the `native` flag) of the `elementPropertyInternal` into a separate function and use it here instead? (not critical though, we can do it in a separate PR if it makes sense)

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](https://test.corp.google.com/ui#id=OCL:258193021:BASE:258193055:1563212519808:f7250b26)
- [Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:258193021:BASE:258193317:1563212589557:4bd7c278)

**Review by kara — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
VE Presubmit is successful, Ivy Presubmit looks normal.

**Comment by kara** [comment]
merge-assistance: global
